lib/evm/precompiles/09-blake2f.ts

```diff
@@ -0,0 +1,438 @@
+  7,
+  1,
+  4,
+  10,
+  5,
+  10,
+  2,
+  8,
+  4,
+  7,
+  6,
+  1,
+  5,
+  15,
+  11,
+  9,
+  14,
+  3,
+  12,
+  13,
+  0,
+  0,
+  1,
+  2,
+  3,
+  4,
+  5,
+  6,
+  7,
+  8,
+  9,
+  10,
+  11,
+  12,
+  13,
+  14,
+  15,
+  14,
+  10,
+  4,
+  8,
+  9,
+  15,
+  13,
+  6,
+  1,
+  12,
+  0,
+  2,
+  11,
+  7,
+  5,
+  3,
+]
+
+// These are offsets into a uint64 buffer.
+// Multiply them all by 2 to make them offsets into a uint32 buffer,
+// because this is Javascript and we don't have uint64s
+const SIGMA82 = new Uint8Array(
+  SIGMA8.map(function(x) {
+    return x * 2
+  }),
+)
+
+export function F(h: Uint32Array, m: Uint32Array, t: Uint32Array, f: boolean, rounds: number) {
+  const v = new Uint32Array(32)
+  let i = 0
+
+  // init work variables
+  for (i = 0; i < 16; i++) {
+    v[i] = h[i]
+    v[i + 16] = BLAKE2B_IV32[i]
+  }
+
+  // 128 bits of offset
+  v[24] = v[24] ^ t[0]
+  v[25] = v[25] ^ t[1]
+  v[26] = v[26] ^ t[2]
+  v[27] = v[27] ^ t[3]
+
+  // last block flag set ?
+  if (f) {
+    v[28] = ~v[28]
+    v[29] = ~v[29]
+  }
+
+  // message words
+  const mw = new Uint32Array(32)
+  // get little-endian words
+  for (i = 0; i < 32; i++) {
+    mw[i] = B2B_GET32(m, 4 * i)
+  }
+
+  // twelve rounds of mixing
+  // uncomment the DebugPrint calls to log the computation
+  // and match the RFC sample documentation
+  // util.debugPrint('          m[16]', m, 64)
+  for (i = 0; i < rounds; i++) {
+    // util.debugPrint('   (i=' + (i < 10 ? ' ' : '') + i + ') v[16]', v, 64)
+    const ri = (i % 10) * 16
+    B2B_G(v, mw, 0, 8, 16, 24, SIGMA82[ri + 0], SIGMA82[ri + 1])
+    B2B_G(v, mw, 2, 10, 18, 26, SIGMA82[ri + 2], SIGMA82[ri + 3])
+    B2B_G(v, mw, 4, 12, 20, 28, SIGMA82[ri + 4], SIGMA82[ri + 5])
+    B2B_G(v, mw, 6, 14, 22, 30, SIGMA82[ri + 6], SIGMA82[ri + 7])
+    B2B_G(v, mw, 0, 10, 20, 30, SIGMA82[ri + 8], SIGMA82[ri + 9])
+    B2B_G(v, mw, 2, 12, 22, 24, SIGMA82[ri + 10], SIGMA82[ri + 11])
+    B2B_G(v, mw, 4, 14, 16, 26, SIGMA82[ri + 12], SIGMA82[ri + 13])
+    B2B_G(v, mw, 6, 8, 18, 28, SIGMA82[ri + 14], SIGMA82[ri + 15])
+  }
+
+  for (i = 0; i < 16; i++) {
+    h[i] = h[i] ^ v[i] ^ v[i + 16]
+  }
+}
+
+export default function(opts: PrecompileInput): ExecResult {
+  assert(opts.data)
+
+  const data = opts.data
+  if (data.length !== 213) {
+    return {
+      returnValue: Buffer.alloc(0),
+      gasUsed: opts.gasLimit,
+      exceptionError: new VmError(ERROR.OUT_OF_RANGE),
+    }
+  }
+  const lastByte = data.slice(212, 213)[0]
+  if (lastByte !== 1 && lastByte !== 0) {
+    return {
+      returnValue: Buffer.alloc(0),
+      gasUsed: opts.gasLimit,
+      exceptionError: new VmError(ERROR.OUT_OF_RANGE),
+    }
+  }
+
+  const rounds = data.slice(0, 4).readUInt32BE(0)
+  const hRaw = data.slice(4, 68)
+  const mRaw = data.slice(68, 196)
+  const tRaw = data.slice(196, 212)
+  // final
+  const f = data.slice(212, 213)[0] === 1
+
+  const gasUsed = new BN(opts._common.param('gasPrices', 'blake2bRound'))
```

Comment:
Nitpick, it should not be `blake2bRound` but `blake2Round`.

Reply:
Created PR: https://github.com/ethereumjs/ethereumjs-common/pull/63